core/src/main/scala/com/mesosphere/usi/core/javadsl/SchedulerAdapter.scala

```diff
@@ -0,0 +1,84 @@
+package com.mesosphere.usi.core.javadsl
+
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import akka.{Done, NotUsed}
+import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult, scaladsl}
+import akka.stream.javadsl.{Flow, Sink, SinkQueueWithCancel, Source, SourceQueueWithComplete}
+import com.mesosphere.usi.core.Scheduler.{SpecInput, StateOutput}
+import com.mesosphere.usi.core.scaladsl.{SchedulerAdapter => ScalaSchedulerAdapter}
+import com.mesosphere.usi.core.models.{SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+
+import scala.compat.java8.FutureConverters._
+import akka.japi.tuple.Tuple3
+import akka.japi.Pair
+
+class SchedulerAdapter(schedulerFlow: scaladsl.Flow[SpecInput, StateOutput, NotUsed], mat: Materializer) {
+
+  private implicit val materializer: Materializer = mat
+
+  private val delegate = new ScalaSchedulerAdapter(schedulerFlow)
+
+  /**
+    * Represents the scheduler as a Sink and Source.
+    *
+    * This method will materialize the scheduler first, then Sink and Source can be materialized independently.
+    *
+    * @param specsSnapshot Snapshot of the current specs
+    * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
+    */
+  def asSourceAndSink(specsSnapshot: SpecsSnapshot = SpecsSnapshot.empty)
+    : Tuple3[CompletableFuture[StateSnapshot], Source[StateEvent, Any], Sink[SpecUpdated, Any]] = {
+    val (snap, source, sink) = delegate.asSourceAndSink(specsSnapshot)
+    Tuple3(snap.toJava.toCompletableFuture, source.asJava, sink.asJava)
+  }
+
+  /**
+    *
+    * Represents the scheduler as a Flow.
+    *
+    *
+    * This method will materialize the scheduler first, then Flow can be materialized independently.
+    * @param specsSnapshot Snapshot of the current specs
+    * @return
+    */
+  def asFlow(specsSnapshot: SpecsSnapshot = SpecsSnapshot.empty)
+    : Pair[CompletableFuture[StateSnapshot], Flow[SpecUpdated, StateEvent, NotUsed]] = {
+    val (snap, flow) = delegate.asFlow(specsSnapshot)
+    Pair(snap.toJava.toCompletableFuture, flow.asJava)
+  }
+
+  /**
+    *
+    * Represents the scheduler as a SourceQueue and a SinkQueue.
```

Comment:
What is the difference to `Sink` and `Source`? We probably should mention this in the docs.

Reply:
Well, I guess akka javadoc will tell it better than me :) Do you still want me to put it there?

Reply:
I agree with @jeschkies  We should do 2 things here.   
1. provide a link to Akka docs for more details on Sink vs Source
2. more important... help bridge the Sink vs Source of this specific domain for the reader of this code.   An example that is context specific to USI would be valuable IMO

Reply:
The usage example will be in the follow-up PR. I think this PR is already too big to include it.